### PR TITLE
Issue 3258: Build fails on JDK11: Bytecode is targeted jdk11 and not jdk8

### LINKED ISSUE #3258: Build fails on JDK11: Bytecode is targeted jdk11 and not jdk8
**Problem description**

Building current master on JDK11 we get this error. It seems that Gradle is trying to parse Java 11 bytecode (from the stacktrace you can see NestMates which is a jdk11 feature) and current version
of Gradle (4.x) does not support it.
The real problem is that we are trying to compile for jdk8 but we are getting java11 bytecode.
 

```
* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':bindings:test'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.run(EventFiringTaskExecuter.java:51)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:46)
        at org.gradle.execution.taskgraph.LocalTaskInfoExecutor.execute(LocalTaskInfoExecutor.java:42)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:277)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:262)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:135)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:130)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.execute(DefaultTaskPlanExecutor.java:200)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.executeWithWork(DefaultTaskPlanExecutor.java:191)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.run(DefaultTaskPlanExecutor.java:130)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.api.GradleException: failed to read class file /home/eolivelli/dev/pravega/bindings/build/classes/java/test/io/pravega/storage/filesystem/FileSystemStorageTest$RollingStorageTests.class
        at org.gradle.api.internal.tasks.testing.detection.AbstractTestFrameworkDetector.classVisitor(AbstractTestFrameworkDetector.java:126)
        at org.gradle.api.internal.tasks.testing.detection.AbstractTestFrameworkDetector.processTestClass(AbstractTestFrameworkDetector.java:145)
        at org.gradle.api.internal.tasks.testing.detection.AbstractTestFrameworkDetector.processTestClass(AbstractTestFrameworkDetector.java:136)
        at org.gradle.api.internal.tasks.testing.detection.DefaultTestClassScanner$1.visitClassFile(DefaultTestClassScanner.java:58)
        at org.gradle.api.internal.tasks.testing.detection.DefaultTestClassScanner$ClassFileVisitor.visitFile(DefaultTestClassScanner.java:76)
        at org.gradle.api.internal.file.collections.jdk7.Jdk7DirectoryWalker$1.visitFile(Jdk7DirectoryWalker.java:86)
        at org.gradle.api.internal.file.collections.jdk7.Jdk7DirectoryWalker$1.visitFile(Jdk7DirectoryWalker.java:59)
        at org.gradle.api.internal.file.collections.jdk7.Jdk7DirectoryWalker.walkDir(Jdk7DirectoryWalker.java:59)
        at org.gradle.api.internal.file.collections.DirectoryFileTree.walkDir(DirectoryFileTree.java:154)
        at org.gradle.api.internal.file.collections.DirectoryFileTree.visitFrom(DirectoryFileTree.java:132)
        at org.gradle.api.internal.file.collections.DirectoryFileTree.visit(DirectoryFileTree.java:117)
        at org.gradle.api.internal.file.collections.FileTreeAdapter.visit(FileTreeAdapter.java:117)
        at org.gradle.api.internal.file.CompositeFileTree.visit(CompositeFileTree.java:93)
        at org.gradle.api.internal.tasks.testing.detection.DefaultTestClassScanner.detectionScan(DefaultTestClassScanner.java:56)
        at org.gradle.api.internal.tasks.testing.detection.DefaultTestClassScanner.run(DefaultTestClassScanner.java:50)
        at org.gradle.api.internal.tasks.testing.processors.TestMainAction.run(TestMainAction.java:55)
        at org.gradle.api.internal.tasks.testing.detection.DefaultTestExecuter.execute(DefaultTestExecuter.java:116)
        at org.gradle.api.internal.tasks.testing.detection.DefaultTestExecuter.execute(DefaultTestExecuter.java:51)
        at org.gradle.api.tasks.testing.AbstractTestTask.executeTests(AbstractTestTask.java:469)
        at org.gradle.api.tasks.testing.Test.executeTests(Test.java:603)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:801)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:768)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
        ... 31 more
Caused by: java.lang.UnsupportedOperationException
        at org.objectweb.asm.ClassVisitor.visitNestHostExperimental(ClassVisitor.java:158)
        at org.objectweb.asm.ClassReader.accept(ClassReader.java:541)
        at org.objectweb.asm.ClassReader.accept(ClassReader.java:391)
        at org.gradle.api.internal.tasks.testing.detection.AbstractTestFrameworkDetector.classVisitor(AbstractTestFrameworkDetector.java:124)
        ... 67 more


```

**Problem location**
Build configuration

**Suggestions for an improvement**
Ensure that even while building on JDK9+ we get java 8 compatible byte code

### PR DESCRIPTION
**Pull request by eolivelli**
**Change log description**  
Force java source and target version to be "8", by setting sourceCompatibility and targetCompatibily after project 'evaluation'.
Current configuration is not working correctly as it is not really applied to every project.

**Purpose of the change**  
Fixes #3258 

**What the code does**  
Change Gradle configuration in order to force Java Bytecode compatibility level

**How to verify it**  
`JAVA_HOME=/path/to/jdk11 ./gradlew build`

### CODE REVIEW

**Comment by eolivelli** [comment]
I am temporary adding openjdk11 to Travis configuration, in order to let reviewers see the effect of the change.
I will drop it once the review passes.

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ language: java
 install: true
 jdk:
  - oraclejdk8
+ - openjdk11
```
Comment (fpj) [inline .travis.yml]: We need to remove this before merging.

File: gradle/java.gradle [hunk 2]
```diff
@@ -16,8 +16,6 @@ plugins.withId('java') {
     apply plugin: 'maven'
 
     compileJava {
-        sourceCompatibility = "8"
```
Comment (fpj) [inline gradle/java.gradle]: Why doesn't it work here, do you know?

Comment (eolivelli) [reply gradle/java.gradle]: I am not a Gradle expert, but in some subproject this value is overridden or is not applied at all.
There are several questions on stackoverflow about this kind of problems with Gradle.
My fix seems to work and it makes sense to me

File: .travis.yml (hunk 1 above)
Comment (eolivelli) [reply .travis.yml]: done

File: build.gradle [hunk 3]
```diff
@@ -60,6 +60,8 @@ allprojects {
 
     gradle.projectsEvaluated {
         tasks.withType(JavaCompile) {
+            sourceCompatibility = "8"
+            targetCompatibility = "8"
             options.compilerArgs << "-Xlint:unchecked" << "-Xlint:deprecation"
```
Comment (eolivelli) [inline build.gradle]: @fpj  I guess this line is here for the same reason that we need to add source/targetCompatibilty at this point
this way we are sure that such options are applied

I took inspiration from this post:
https://stackoverflow.com/questions/21028438/gradle-sourcecompatibility-has-no-effect-to-subprojects

**Comment by eolivelli** [comment]
@fpj @tkaitchuck 
this patch is good to go
I still have some test to fix on jdk11 then we are done
but this patch can be merged as it is

**Comment by eolivelli** [comment]
@tkaitchuck see https://github.com/pravega/pravega/pull/3262 for the fix for the TLS tests

**Comment by eolivelli** [comment]
Ping

**Comment by eolivelli** [comment]
Can we merge this one so that I can rebase the next and we will make the build pass on jdk11 .

The last part in order to really run Pravega on jdk11 is the upgrade of Netty and Bookkeeper, than there is some possibility of success

Thanks

**Comment by eolivelli** [comment]
@tkaitchuck  @fpj can you please commit this patch ?
It is a blocker for JDK11 support

**Pull request merged**